File: BaseballGame/BaseBallGame.js
/**
 * @param {string[]} ops
 * @return {number}
 */
var calPoints = function (ops) {
  let stack = [];
  ops.forEach((op) => {
    if (parseInt(op)) {
      stack.push(parseInt(op));
    } else if (op === "+") {
      stack.push(
        (stack[stack.length - 1] || 0) + (stack[stack.length - 2] || 0)
      );
    } else if (op === "D") {
      stack.push((stack[stack.length - 1] || 0) * 2);
    } else if (op === "C") {
      stack.pop();
    }
  });
  return stack.reduce((a, b) => a + b, 0);
};

File: BaseballGame/BaseballGame.py
class Solution:
    def calPoints(self, ops: List[str]) -> int:
        stack = []
        count = 0
        for i in ops:
            if i == "C":
                count -= stack.pop()
            elif i == "D":
                stack.append(stack[-1] * 2)
                count += stack[-1]
            elif i == "+":
                stack.append(stack[-1] + stack[-2])
                count += stack[-1]
            else:
                stack.append(int(i))     
                count += stack[-1]
        return count


